etcdutl/snapshot/v3_snapshot_test.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2018 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snapshot
+
+import (
+	"context"
+	"errors"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"go.etcd.io/bbolt"
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/server/v3/embed"
+	"go.etcd.io/etcd/server/v3/etcdserver"
+	"go.etcd.io/etcd/server/v3/storage/mvcc"
+	"go.etcd.io/etcd/server/v3/storage/schema"
+)
+
+// TestSnapshotStatus is the happy case.
+// It inserts pre-defined number of keys and asserts the output hash of status command.
+// The expected hash value must not be changed.
+// If it changes, there must be some backwards incompatible change introduced.
```

Comment:
Could we merge the tests before we change logic? It's nice to separate the refactors and tests that are meant to prove nothing change.

Reply:
I don't think that's necessary because refactors doesn't affect the hash value in any way. They're only about adding extra error conditions.